plutus-ledger-api/src/Plutus/V1/Ledger/Api.hs

```diff
@@ -185,7 +193,7 @@ evaluateScriptRestricting verbose cmdata budget p args = swap $ runWriter @LogOu
     let (res, _, logs) =
             UPLC.runCek
                 (toMachineParameters model)
-                (UPLC.restricting $ PLC.ExRestrictingBudget budget)
+                (UPLC.restricting $ PLC.ExRestrictingBudget (scaleUp costScaleFactor budget))
```

Comment:
We now have `ExBudget` values involving different units, so I'd really like to expose a type different from `ExBudget` here for use by the ledger and have `toExBudget` and `fromExBudget` functions to convert the units along with the types.  That would change the interface though, which might be annoying. (Also, what would we call the new units?)

Reply:
I guess so, but there might be a risk of someone importing the wrong thing. I vaguely thought about adding a phantom type parameter to the `ExMemory` type that we could use to say whether it contained ledger units or internal units.  I don't know whether that's sensible or not.

Reply:
Well, if they import the wrong thing it won't compile, right? Since we'll insist on being passed the type that we export to the ledger and not the internal one?

But let's do this in another PR anyway.

Reply:
> I vaguely thought about adding a phantom type parameter to the `ExMemory` type that we could use to say whether it contained ledger units or internal units. I don't know whether that's sensible or not.

I like the idea.

Reply:
> Well, if they import the wrong thing it won't compile, right? 

I think I was worrying about people poking about and using things outside `Api.hs` without realising that the units were different.  But yes, we could have another small PR later to make things a bit more type-safe.